x-pack/elastic-agent/pkg/agent/transpiler/rules.go

```diff
@@ -1519,6 +1523,107 @@ func InsertDefaults(path string, selectors ...Selector) *InsertDefaultsRule {
 	}
 }
 
+type InjectQueueRule struct{}
+
+// InjectQueue creates a InjectQueueRule
+func InjectQueue() *InjectQueueRule {
+	return &InjectQueueRule{}
+}
+
+func (r *InjectQueueRule) Apply(agentInfo AgentInfo, ast *AST) (err error) {
+	defer func() {
+		if err != nil {
+			err = errors.New(err, "failed to inject queue parameters into configuration")
+		}
+	}()
+
+	outputNode, found := Lookup(ast, "output")
+	if !found {
+		return nil
+	}
+
+	outputDict, ok := outputNode.Value().(*Dict)
+	if !ok || len(outputDict.value) == 0 {
+		faeNode := &Key{
+			name:  "hello",
+			value: &StrVal{value: "fae was here"},
+		}
+		if err := Insert(ast, faeNode, "output"); err != nil {
```

Comment:
Should this be here at all? Was this a debugging thing?

I think if no output is defined it should just return nil and do nothing.

Reply:
Haha yes, leftover from when I was still figuring out how to traverse ASTs, removed